pushnotifications/src/test/java/com/pusher/pushnotifications/internal/PersistentJobQueueTest.kt

```diff
@@ -17,52 +20,194 @@ class PersistentJobQueueTest {
+    val queue: PersistentJobQueue<ServerSyncJob> = TapeJobQueue(tempFile, converter)
+    queue.push(UnsubscribeJob("interest_unsubscribe"))
+
+    val returnedList = queue.asIterable().toList()
+
+    assertNotNull(returnedList)
+    assertEquals(1, returnedList.size)
+
+    assertEquals(UnsubscribeJob::class.java, returnedList[0].javaClass)
+    val unsubscribeJob:UnsubscribeJob = returnedList[0] as UnsubscribeJob
+    assertEquals("interest_unsubscribe", unsubscribeJob.interest)
+  }
+
+  @Test
+  fun `iterables returns correct Set Subscriptions Job`() {
+    val queue: PersistentJobQueue<ServerSyncJob> = TapeJobQueue(tempFile, converter)
+    queue.push(SetSubscriptionsJob(setOf("subscribe_one", "subscribe_two")))
+
+    val returnedList = queue.asIterable().toList()
+
+    assertNotNull(returnedList)
+    assertEquals(1, returnedList.size)
+
+    assertEquals(SetSubscriptionsJob::class.java, returnedList[0].javaClass)
+    val unsubscribeJob:SetSubscriptionsJob = returnedList[0] as SetSubscriptionsJob
+    assertEquals("subscribe_one", unsubscribeJob.interests.first())
+    assertEquals("subscribe_two", unsubscribeJob.interests.last())
+  }
+
+  @Test
+  fun `iterables returns correct Application Start Job`() {
+    val queue: PersistentJobQueue<ServerSyncJob> = TapeJobQueue(tempFile, converter)
+    queue.push(ApplicationStartJob(
+            DeviceMetadata("sdk_version", "android_version")))
+
+    val returnedList = queue.asIterable().toList()
+
+    assertNotNull(returnedList)
+    assertEquals(1, returnedList.size)
+
+    assertEquals(ApplicationStartJob::class.java, returnedList[0].javaClass)
+    val applicationStartJob:ApplicationStartJob = returnedList[0] as ApplicationStartJob
+    assertEquals("sdk_version", applicationStartJob.deviceMetadata.sdkVersion)
+    assertEquals("android_version", applicationStartJob.deviceMetadata.androidVersion)
+  }
+
+  @Test
+  fun `iterables returns correct Application Set User Id Job`() {
+    val queue: PersistentJobQueue<ServerSyncJob> = TapeJobQueue(tempFile, converter)
+    queue.push(SetUserIdJob("user_id"))
+
+    val returnedList = queue.asIterable().toList()
+
+    assertNotNull(returnedList)
+    assertEquals(1, returnedList.size)
+
+    assertEquals(SetUserIdJob::class.java, returnedList[0].javaClass)
+    val setUserIdJob:SetUserIdJob = returnedList[0] as SetUserIdJob
+    assertEquals("user_id", setUserIdJob.userId)
+  }
+
+  @Test
+  fun `iterables returns correct Server Sync Job Types`() {
+
+    val queue: PersistentJobQueue<ServerSyncJob> = TapeJobQueue(tempFile, converter)
+
+    queue.push(StartJob("fcm_token", arrayListOf("one", "two")))
+    queue.push(RefreshTokenJob("new_token"))
+    queue.push(SubscribeJob("interest_subscribe"))
+    queue.push(UnsubscribeJob("interest_unsubscribe"))
+    queue.push(SetSubscriptionsJob(setOf("subscribe_one", "subscribe_two")))
+    queue.push(ApplicationStartJob(
+            DeviceMetadata("sdk_version", "android_version")))
```

Comment:
I personally don't like these values as they are not realistic examples of what they will hold when used by customers. I get that using vegetables is also not 100% realistic, but at least it's using them more closely as customers would think about them. This is more in line with the principle of "the closer test-data matches production use, the better".

In this case, the sdk version would never hold a value like `"sdk_version"` which just makes it read weird to me. Anyway, this is all a bit subjective, so treat it as an optional nitpick. :)